rmw_connext_shared_cpp/src/shared_functions.cpp

```diff
@@ -510,6 +510,185 @@ destroy_guard_condition(const char * implementation_identifier,
   return result;
 }
 
+rmw_waitset_t *
+create_waitset(const char * implementation_identifier,
+  rmw_guard_conditions_t * fixed_guard_conditions, size_t max_conditions)
+{
+  rmw_waitset_t * waitset = rmw_waitset_allocate();
+
+  ConnextWaitSetInfo * waitset_info = nullptr;
+  DDSGuardCondition * dds_guard_cond = nullptr;
+  DDS_ReturnCode_t ret;
+
+  // From here onward, error results in unrolling in the goto fail block.
+  if (!waitset) {
+    RMW_SET_ERROR_MSG("failed to allocate waitset");
+    goto fail;
```

Comment:
In this case `waitset_info` is a `nullptr` but in the first line under `fail` `waitset_info->active_conditions` is being accessed.